.git_hooks/pre-commit

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env sh
+if ! cargo fmt -- --check ; then
+  printf "\n"
+  printf "\033[0;31mpre-commit hook failed during:\033[0m\n"
+  printf "\033[0;31m\tcargo fmt -- --check\033[0m\n"
+  exit 1
+fi
+
+if ! cargo clippy --locked -- -D warnings ; then
+  printf "\n"
+  printf "\033[0;31mpre-commit hook failed during:\033[0m\n"
+  printf "\033[0;31m\tclippy --locked -- -D warning\033[0m\n"
+  exit 1
+fi
+
```

Comment:
No `cargo test`?

Reply:
`cargo test` is somewhat slow, maybe we should put it in `pre-push` and not `pre-commit`?

Reply:
Ooh, I like that idea :smile: ignore my later comment about removing the `.git_hooks` dir then.

Reply:
Already working on its PR 😄

Reply:
Feel free to add it to this one, it's still part of the pre-commit script :)

Reply:
Preferred to create a separate PR https://github.com/rust-lang/docs.rs/pull/977.
I think we can merge this one, I don't see any pending comment.